### PR TITLE
Last operation time

### PR DESCRIPTION
**Pull request by sergeysenja1992**

### CODE REVIEW

**Review by dtsarenko — APPROVED** [review]

File: src/main/java/com/icthh/xm/ms/balance/service/BalanceService.java [hunk 1]
```diff
@@ -314,6 +302,31 @@ public TransferDto transfer(TransferBalanceRequest transferRequest) {
             .build();
     }
 
+    private BalanceChangeEvent createBalanceChangeEvent(String operationUuid, BigDecimal amount, Metadata metadata,
+                                                   Instant now, Balance balance, boolean isSubtractAmount,
+                                                   OperationType transferTo) {
+        BigDecimal amountBeforeTransferTo = balanceRepository.findBalanceAmount(balance).orElse(ZERO);
+        BigDecimal amountAfterTransferTo = getAmountAfter(isSubtractAmount, amountBeforeTransferTo, amount);
+        Optional<BalanceChangeEvent> lastBalanceChangeEvent = balanceChangeEventRepository.findLastBalanceChangeEvent(balance.getId());
```
Comment (amedvedchuk) [inline src/main/java/com/icthh/xm/ms/balance/service/BalanceService.java]: seems you do not need separate variable `lastBalanceChangeEvent`, so do not break optional processing chain (find().map().orElse()).

Comment (sergeysenja1992) [reply src/main/java/com/icthh/xm/ms/balance/service/BalanceService.java]: but, it's this way more readable

Comment (amedvedchuk) [reply src/main/java/com/icthh/xm/ms/balance/service/BalanceService.java]: Do you really think that having the intermediate variable `lastBalanceChangeEvent` is more readable than that : ?
```
Instant prevOperationDate = balanceChangeEventRepository
      .findLastBalanceChangeEvent(balance.getId())
      .map(BalanceChangeEvent::getOperationDate)
      .orElse(Instant.EPOCH);
```

Comment (sergeysenja1992) [reply src/main/java/com/icthh/xm/ms/balance/service/BalanceService.java]: Fixed

**Review by VovaViter — APPROVED** [review]

**Review by amedvedchuk — APPROVED** [review]

**Pull request merged**